src/taipy/core/data/_data_model.py

```diff
@@ -74,3 +91,177 @@ def from_dict(data: Dict[str, Any]):
             edit_in_progress=bool(data.get("edit_in_progress", data.get("edition_in_progress", False))),
             data_node_properties=dn_properties,
         )
+
+    @classmethod
```

Comment:
@joaoandre-avaiga the reason I chose entity and not entity model to keep the business logic is that some of the logic is specific to certain child entities. The clearest example (probably the only example as well) is DataNodes. There are some logic that only works for SQL or generic for example, while some work for a group of child entities like csv, excel, SQL table, parquet (the tabular DNs). So this helps break down the logic even further based on the entities. But the model always works for me, I'm just not sure I understand it when you said "not always we will have access to the entity from the repositories." To me, the repo has access to both, no? I didn't have much trouble with that so :/

Reply:
Unfortunatelly the repository does not have access to the entity on most of it's methods. For instance, take the load method:
```python
    def _load(self, model_id: str) -> Entity:
        try:
            model = self.model(**self._get(self.__get_model_filepath(model_id)))
            return self.converter._model_to_entity(model)
        except FileNotFoundError:
            raise ModelNotFound(str(self.dir_path), model_id)
 ```
 As you can see, the repository has no way of knowing which entity it has to return. So that's where the converter enters. Maybe we could inject the entity in the repository, in the same manner as we do the converter. It could be a design choice.

What do you think @jrobinAV ?